verifiers/internal/gha/provenance.go

```diff
@@ -42,22 +42,14 @@ func provenanceFromEnv(env *dsselib.Envelope) (prov *intoto.ProvenanceStatement,
 }
 
 // Verify Builder ID in provenance statement.
-func verifyBuilderID(prov *intoto.ProvenanceStatement, builderID string) error {
-	// Check that the BuilderID is well-formed.
-	provid, err := sourceFromURI(prov.Predicate.Builder.ID, false)
-	if err != nil {
-		return err
-	}
-	// Note: builderID does not contain the tag.
-	// TODO(#189): support cases where user wants to match on the full builderID, including the tag.
-	bid, err := sourceFromURI(builderID, true)
-	if err != nil {
-		return err
-	}
-	if provid != bid {
+// This function does an exact comparison, and expects certBuilderID to be the full
+// `name@refs/tags/<name>`.
+func verifyBuilderID(prov *intoto.ProvenanceStatement, certBuilderID string) error {
```

Comment:
could you change the name to something like `verifyBuilderIDExactMatch`?

Reply:
+1, done.